### DISCUSSION TITLE
A good way to represent URL hash in a locale.

### DISCUSSION THREAD

**Discussion by hochan222** [mdn/mdn-community#332] [Localization] [answered]
## Issue 

Recently, I realized that most locale pages, including the ko locale, do not represent hash well.

`parameters` is being translated as `매개변수` in the ko locale. However, link to `매개변수` is ```[`TypedArray`](/ko/docs/Web/JavaScript/Reference/Global_Objects/TypedArray#parameters)```. On the TypedArray page, it is translated h2 tag title as `## 매개변수`, but the hash of the link is `#parameters`.

With this problem, when you go to a page through a link, you go to the top of the page, not to the corresponding hash. This is because `parameters` has been translated from the page and no longer exists.

## Other locale status

In localization pages, most hashes use `en-us` as it is and is not properly linked. 

For better understanding, some examples are given below.

Language | Translation Word | Origin (en-us) | Translation Word Link
-- | -- | -- | --
fr | valeur_retournée | return_value | `[Valeur retournée](/fr/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Return_value)`
es | objetos_typedarray | TypedArray_objects | `[TypedArray objects](/es/docs/Web/JavaScript/Reference/Global_Objects/TypedArray#TypedArray_objects)`
ja | オブジェクト | TypedArray_objects | `[typed array](/ja/docs/Web/JavaScript/Reference/Global_Objects/TypedArray#TypedArray_objects)`
pt-br | Síntaxe | Syntax | `[TypedArray](/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/TypedArray#Syntax)`
ru | объекты_typedarray | TypedArray_objects | `[typed array](/ru/docs/Web/JavaScript/Reference/Global_Objects/TypedArray#TypedArray_objects)`
zh-cn | 对象 | TypedArray_objects | `[typed array](/zh-cn/docs/Web/JavaScript/Reference/Global_Objects/TypedArray#TypedArray_objects)`
zh-tw | 物件 | TypedArray_objects | `[typed array](/zh-tw/docs/Web/JavaScript/Reference/Global_Objects/TypedArray#TypedArray_objects)`

## Worrying point

I wonder if it would be good to specify hash based on the word translated to the locale rather than based on `en-us`. Even if we write a [glossary guide](https://github.com/hochan222/translated-content/blob/ko-guide-translate/docs/ko/guides/glossary-guide.md) for consistent titles, there are also titles that are sentences rather than words. Do we have to specify all sentence or phrase titles? Maintaining the correct hash value is a big cost in the current state.

There is also the problem of encoding.

In common, is there any way to manage it with a consistent syntax across all locales? or another good solution.

**Comment by yin1999** [comment]
I l10n-zh, we just use the translated hash, we could get the hash by clicking the heading and copying it from the address bar (or get it from dev tools, for h4, h5, etc.)

![image](https://user-images.githubusercontent.com/15844309/219845985-cfb63d76-3903-4b72-9893-cf382a3756ed.png)

But with localized hash, we could not go to the corresponding section, when we switch the locale.

**Reply by hochan222** [reply]
Thank you for your answer. The issue you mentioned is also a problem.

One more thing I want to point out is that hash access is being made to a specific page from another page, not a problem within one page.

**Reply by yin1999** [reply]
>  want to point out is that hash access is being made to a specific page from another page, not a problem within one page

Yes, we have to check the hash when we pointing to a section in localized pages for now.

**Reply by pransh15** [reply]
Hi folks, I just wanted to check if this has been answered and if we can close this.

**Reply by hochan222** [reply, marked as answer]
@pransh15, Hi! 

The discussion below covers the above in more detail, so I will close this discussion for now. Thank you :) 

https://github.com/orgs/mdn/discussions/729

**Reply by pransh15** [reply]
Perfect, thanks @hochan222! 🙌
